nvm.sh

```diff
@@ -217,6 +217,28 @@ nvm_rc_version() {
   nvm_echo "Found '${NVMRC_PATH}' with version <${NVM_RC_VERSION}>"
 }
 
+nvm_find_package_json() {
+  local dir
+  dir="$(nvm_find_up 'package.json')"
+  if [ -e "${dir}/package.json" ]; then
+    nvm_echo "${dir}/package.json"
+  fi
+}
+
+nvm_package_json_semver_expression() {
+  export PKG_JSON_SEMVER_EXPRESSION=''
+  local PKG_JSON_PATH
+  PKG_JSON_PATH="$(nvm_find_package_json)"
+  if [ ! -e "${PKG_JSON_PATH}" ]; then
+    nvm_err "No package.json file found"
+    return 1
+  fi
+  # TODO handle error case
+  # TODO implement without node?
```

Comment:
Oh, I definitely think it should eventually be in `install` as well; but either way the semver parsing and range logic should be completely offline.

Yes, every part of `nvm` needs to be implemented without node, full stop.